From ceb7cb855f2321fd8b2f4024dd6b01bf0a2514fb Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Tue, 15 Dec 2020 21:42:14 +0200
Subject: [PATCH] [SW-0] gaudi: Enable iatu access only in secured mode

Enable the iATU access only in secured devices, otherwise
device might stay stuck in that mode if LKD or fw-loader
doesn't disable it

Change-Id: I624186202770d673e9839736f92e08b54f1c436d
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 095a2c7..22366de 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -363,8 +363,12 @@
 
 void preboot_conf_iatu_access(bool enable)
 {
-	u32 reg = enable ? 0x00300000 : 0;
+	u32 reg;
 
+	if (!preboot_security_is_enabled())
+		return;
+
+	reg = enable ? 0x00300000 : 0;
 	regmap_writel(reg, mmPCIE_AUX_DBI);
 }
 
